WordPress/gutenberg
# List view: Modify the shortcut to focus while open

## What?


Fixes #29466

This PR does the following.
1. Changes the shortcut to focus the list view when the list view is open and focus is not inside the sidebar.
2. When the list view is open and focus is inside the sidebar, using the shortcut will close it as normal.

## Why?


See the related issue for more information. This allows the list view to remain visible for the sighted and still allows keyboard users to navigate back to the list view without closing and re-opening.

## How?


It is not a pretty solution but it works. I left comments in the code to guide. The rest is magic.

## Testing Instructions




1. This is only in the post editor for now so open the post editor.
2. Use the list view shortcut, on Windows, it is alt+shift+o.
3. Notice how the list view sidebar opens and focus is placed inside.
4. Press the shortcut again.
5. Notice how focus went back to the position from which you opened the list view.
6. Press the shortcut again and list view will open.
7. Select a block.
8. Press the shortcut again.
9. Notice how focus goes back to the currently selected block in the list view.

## Additional thoughts
- [ ] Bug: The `useFocusReturn` hook needs some refactoring or different usage for this. Since it attaches when the component mounts, it will not return focus to where the user expects if they have navigated around the list view for a while. I need some way to call the hook dynamically but my research is turning up nothing. Suggestions on how to fix this would be appreciated.

It seems it would be better to give the sidebars it's own keyboard shortcuts. That would make abstraction of future functionality easier as well. For now, I think I'll leave it be. Being able to pass things from parent to child would likely help but we're not setup for it now.

## Screenshots or screencast
